updater

```diff
@@ -515,45 +516,53 @@ update_now_cli() { #input: updatable_files and updatable_apps variables
   cp -a "${DIRECTORY}/update/pi-apps/.git" "${DIRECTORY}" || error "Failed to copy new .git folder!"
   
   [ "$no_status" != true ] && status_green "\nPi-Apps updates complete."
+  true
 }
 
-update_now_gui_apps() { # updatable_apps variable
-  # this function exists to avoid having to write commands in terminal-run formatting for update_now_gui
-  local failed_apps=""
+update_now_gui() { #input: updatable_files and updatable_apps variables
   local IFS=$'\n'
-  for app in $updatable_apps ;do
-    update_app "$app" || failed_apps+="$app"$'\n'
-  done
-  
-  #Set terminal title
-  echo -ne "\e]0;Updates complete\a"
-  
-  #load the app list now to reduce launch time
-  refresh_app_list &
-  
-  action=update diagnose_apps "$failed_apps"
-}
+  local queue
+  safe_wait() {
+    # terminal_manage (which calls etc/terminal-run) is sensitive to race conditions. If terminal_manage is called before the first terminal-run terminal opens, multiple terminals will open.
+    # the for loop is very fast and without a sleep would result in the number of terminals equaling the number of updatable_apps which is not the intention.
+    # the intention of terminal_manage is for there to only be one terminal that all manage executions use sequentially
+    #Use a PID file to check if another daemon process is running.
+    if [ -f "${DIRECTORY}/data/manage-daemon/pid" ] && [ -f "/proc/$(cat "${DIRECTORY}/data/manage-daemon/pid")/status" ];then
+      true #nothing to do
+    else
+      attempts=0
+      while [ ! -f "${DIRECTORY}/data/manage-daemon/pid" ] || [ ! -f "/proc/$(cat "${DIRECTORY}/data/manage-daemon/pid")/status" ];do
+        if [ "$attempts" -gt 500 ];then
+          break
+        fi
+        sleep 0.01
+        attempts=$((attempts + 1))
+      done
+    fi
+  }
+  #update any files with terminal_manage. if it fails terminal-run will update the files in the background. This is a safety measure to update pi-apps scripts if terminal fails to launch.
+  if [ ! -z "$updatable_files" ]; then
+    queue+="$(echo "$updatable_files" | sed 's/^/update_file /g')"$'\n'
```

Comment:
this will add a file as a new line in the queue for each individual file which is what I was trying to avoid. running manage takes about 2 seconds minimum per action regardless of what you do (due to sourcing api and other things)
the input:
```
file1
file2
file3
```
```
update_file file1
update_file file2
update_file file3
```